pkg/reconciler/ingress/ingress.go

```diff
@@ -166,6 +170,406 @@ func SetupSecretTracker(ctx context.Context, cmw configmap.Watcher, init Reconci
 // converge the two. It then updates the Status block of the Ingress resource
 // with the current status of the resource.
 func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
-	// TBD
+	return ReconcileIngress(r.ConfigStore.ToContext(ctx), r, key)
+}
+
+// ReconcileIngress retrieves Ingress by key and performs reconciliation
+func ReconcileIngress(ctx context.Context, ra ReconcilerAccessor, key string) error {
```

Comment:
I wonder if we could further reduce the surface of `ReconcilerAccessor` if this has `(bir *BaseIngressReconciler)` as the receiver?

Reply:
Done.